proxy/src/main/java/com/velocitypowered/proxy/protocol/packet/Chat.java

```diff
@@ -10,20 +10,25 @@
 import net.kyori.text.serializer.gson.GsonComponentSerializer;
 import org.checkerframework.checker.nullness.qual.Nullable;
 
+import java.util.UUID;
+
 public class Chat implements MinecraftPacket {
 
   public static final byte CHAT_TYPE = (byte) 0;
   public static final int MAX_SERVERBOUND_MESSAGE_LENGTH = 256;
+  public static final UUID EMPTY_SENDER = new UUID(0, 0);
 
   private @Nullable String message;
   private byte type;
+  private @Nullable UUID sender;
 
   public Chat() {
   }
 
-  public Chat(String message, byte type) {
+  public Chat(String message, byte type, UUID sender) {
```

Comment:
I should probably add a dummy constructor here that has message and type

Reply:
The packet API is internal details. You can create a constructor for compatibility if you want, but it won't prevent me from merging this in.